src/context/index.ts

```diff
@@ -161,15 +164,19 @@ export class Context {
     }
 
     if (currentPair) {
-      return new Context({
+      context = new Context({
         polymeshApi,
         middlewareApi,
         keyring,
         pair: currentPair,
       });
+    } else {
+      context = new Context({ polymeshApi, middlewareApi, keyring });
     }
 
-    return new Context({ polymeshApi, middlewareApi, keyring });
+    context.isArchiveNode = await context.isCurrentNodeArchive();
```

Comment:
Having `context.isCurrentNodeArchive()` as a public function along `context.isArchiveNode` is confusing. You should just make the relevant calls here and delete the function

Reply:
The comments from the `isCurrentNodeArchive` function should be adapted and used for the `isArchiveNode` variable